pkg/slayers/path/scion/raw_spec.gobra

```diff
@@ -192,12 +207,26 @@ requires 0 <= idx && idx <= unfolding acc(r.Mem(ubuf), p) in len(r.Raw)
 ensures  acc(&r.Raw, p/2)
 ensures  r.Raw === old(unfolding acc(r.Mem(ubuf), p) in r.Raw)
 ensures  acc(slices.AbsSlice_Bytes(r.Raw[:idx], 0, idx), p) && acc(r.Base.Mem(), p/2)
+ensures  forall i int :: 0 <= i && i < idx ==> (unfolding acc(slices.AbsSlice_Bytes(r.Raw[:idx], 0, idx), p) in &r.Raw[:idx][i] == &r.Raw[i] && r.Raw[i] == old(unfolding acc(r.Mem(ubuf), p) in unfolding acc(slices.AbsSlice_Bytes(ubuf, 0, len(ubuf)), p) in r.Raw[i]))
+ensures  (unfolding acc(r.Base.Mem(), p/2) in r.Base.PathMeta.CurrINF) == old(unfolding acc(r.Mem(ubuf), p) in unfolding acc(r.Base.Mem(), p) in r.Base.PathMeta.CurrINF)
+ensures  (unfolding acc(r.Base.Mem(), p/2) in r.Base.PathMeta.SegLen[0]) == old(unfolding acc(r.Mem(ubuf), p) in unfolding acc(r.Base.Mem(), p) in r.Base.PathMeta.SegLen[0])
+ensures  (unfolding acc(r.Base.Mem(), p/2) in r.Base.PathMeta.SegLen[1]) == old(unfolding acc(r.Mem(ubuf), p) in unfolding acc(r.Base.Mem(), p) in r.Base.PathMeta.SegLen[1])
+ensures  (unfolding acc(r.Base.Mem(), p/2) in r.Base.PathMeta.SegLen[2]) == old(unfolding acc(r.Mem(ubuf), p) in unfolding acc(r.Base.Mem(), p) in r.Base.PathMeta.SegLen[2])
+ensures  (unfolding acc(r.Base.Mem(), p/2) in r.Base.NumINF) == old(unfolding acc(r.Mem(ubuf), p) in unfolding acc(r.Base.Mem(), p) in r.Base.NumINF)
```

Comment:
these lines are extremely long and hard to read. we can split this in multiple lines and, above all, introduce pure functions (e.g., one that takes an index `i` between 0 and 2 and returns `r.Base.PathMeta.SegLen[i]`)